.github/workflows/test.yml

```diff
@@ -2,6 +2,9 @@ name: CI
 
 on:
   push:
+    branches:
```

Comment:
This ensures that it runs only once per PR, but always runs on direct pushes to `dev` and `main` branch